File: src/frontend/src/types/target_profile.ts
export interface TargetProfileCreate {
  name: string
  department_id: string
  position?: string
  description?: string
}

export interface TargetProfileUpdate {
  name?: string
  position?: string
  description?: string
}

export interface TargetProfileCompetencyInput {
  competency_id: string
  required_level: number
  is_mandatory?: boolean
}

export interface CompetencyBrief {
  id: string
  name: string
}

export interface TargetProfileCompetencyRead {
  competency_id: string
  required_level: number
  is_mandatory: boolean
  competency: CompetencyBrief
}

export interface DepartmentBrief {
  id: string
  name: string
}

export interface TargetProfileRead {
  id: string
  name: string
  department_id: string
  department: DepartmentBrief
  position: string | null
  description: string | null
  competencies: TargetProfileCompetencyRead[]
  created_at: string
  updated_at: string
}

export interface GapItem {
  competency_id: string
  competency_name: string
  required_level: number
  is_mandatory: boolean
  current_score: number | null
}

File: src/backend/app/schemas/target_profile.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class TargetProfileCompetencyInput(BaseModel):
    competency_id: uuid.UUID
    required_level: int
    is_mandatory: bool = False


class TargetProfileCreate(BaseModel):
    name: str
    department_id: uuid.UUID
    position: str | None = None
    description: str | None = None


class TargetProfileUpdate(BaseModel):
    name: str | None = None
    position: str | None = None
    description: str | None = None


class CompetencyBrief(BaseModel):
    id: uuid.UUID
    name: str
    model_config = {"from_attributes": True}


class TargetProfileCompetencyRead(BaseModel):
    competency_id: uuid.UUID
    required_level: int
    is_mandatory: bool
    competency: CompetencyBrief
    model_config = {"from_attributes": True}


class DepartmentBrief(BaseModel):
    id: uuid.UUID
    name: str
    model_config = {"from_attributes": True}


class TargetProfileRead(BaseModel):
    id: uuid.UUID
    name: str
    department_id: uuid.UUID
    department: DepartmentBrief
    position: str | None
    description: str | None
    competencies: list[TargetProfileCompetencyRead] = []
    created_at: datetime
    updated_at: datetime
    model_config = {"from_attributes": True}
